core/src/main/java/com/google/cloud/sql/core/CloudSqlInstance.java

```diff
@@ -119,33 +96,22 @@ class CloudSqlInstance {
    * Initializes a new Cloud SQL instance based on the given connection name.
    *
    * @param connectionName instance connection name in the format
-   *     "PROJECT_ID:REGION_ID:INSTANCE_ID"
-   * @param apiClient Cloud SQL Admin API client for interacting with the Cloud SQL instance
+   * "PROJECT_ID:REGION_ID:INSTANCE_ID"
+   * @param apiService Service class for interacting with the Cloud SQL Admin API
    * @param executor executor used to schedule asynchronous tasks
    * @param keyPair public/private key pair used to authenticate connections
    */
   CloudSqlInstance(
       String connectionName,
-      SQLAdmin apiClient,
+      SqlAdminApiService apiService,
       boolean enableIamAuth,
       CredentialFactory tokenSourceFactory,
       ListeningScheduledExecutorService executor,
       ListenableFuture<KeyPair> keyPair) throws IOException, InterruptedException {
 
```

Comment:
All these extracts should be one PR at a time -- we can either do them when introducing the new associated class, or introduce the class in one PR, and then use it in another.